### PR TITLE
take apart dependent image tags

### PR DESCRIPTION
**Pull request by gitlawr**
Currently, we are using the same tag to build pipeline images. But the versions of dependent images don't have to be the same, so it would be better to take the image tags apart.

### CODE REVIEW

File: scripts/package [hunk 1]
```diff
@@ -15,12 +15,18 @@ cp ../bin/pipeline ./pipeline/
...
+docker build -t ${REPO}/jenkins-boot:${JENKINS_BOOT_TAG} ./jenkins-boot
 
-for Folder_a in ${Folders}/*; do
-  Folder_a_name=`echo "$Folder_a" | awk -F '\/' '{print $NF}'`
-  docker build -t ${REPO}/${Folder_a_name}:${TAG} $Folder_a/.
-  echo ${REPO}/${Folder_a_name}:${TAG} >> $(dirname $0)/../dist/images
-done
-cat $(dirname $0)/../dist/images
+echo built images:
+echo ${REPO}/pipeline:${TAG}
```
Comment (orangedeng) [inline scripts/package]: It should place the `echo ...` after each docker build

File: scripts/package [hunk 2]
```diff
@@ -15,12 +15,18 @@ cp ../bin/pipeline ./pipeline/
...
+echo ${REPO}/pipeline-ui:${PIPELINE_UI_TAG}
```
Comment (orangedeng) [inline scripts/package]: ^^

File: scripts/package [hunk 3]
```diff
@@ -15,12 +15,18 @@ cp ../bin/pipeline ./pipeline/
...
+echo ${REPO}/pipeline-jenkins-slave:${JENKINS_SLAVE_TAG}
```
Comment (orangedeng) [inline scripts/package]: ^^

File: scripts/package [hunk 4]
```diff
@@ -15,12 +15,18 @@ cp ../bin/pipeline ./pipeline/
...
+echo ${REPO}/jenkins-boot:${JENKINS_BOOT_TAG}
```
Comment (orangedeng) [inline scripts/package]: ^^

File: scripts/package [hunk 5]
```diff
@@ -15,12 +15,18 @@ cp ../bin/pipeline ./pipeline/
 
 TAG=${TAG:-${VERSION}${SUFFIX}}
 REPO=${REPO:-rancher}
+#Tag for dependent images
+PIPELINE_UI_TAG=v0.1.1
```
Comment (biblesyme) [inline scripts/package]: Shall we keep the tag name same as its repository? Like, in Pipeline UI, its tag is v1.0.1?

File: scripts/package [hunk 6]
```diff
@@ -15,12 +15,18 @@ cp ../bin/pipeline ./pipeline/
...
-Folders="$(dirname $0)/../package"
+docker build -t ${REPO}/pipeline:${TAG} ./pipeline
+docker build -t ${REPO}/pipeline-ui:${PIPELINE_UI_TAG} ./pipeline-ui
+docker build -t ${REPO}/pipeline-jenkins-slave:${JENKINS_SLAVE_TAG} ./pipeline-jenkins-slave
+docker build -t ${REPO}/jenkins-boot:${JENKINS_BOOT_TAG} ./jenkins-boot
 
-for Folder_a in ${Folders}/*; do
-  Folder_a_name=`echo "$Folder_a" | awk -F '\/' '{print $NF}'`
-  docker build -t ${REPO}/${Folder_a_name}:${TAG} $Folder_a/.
-  echo ${REPO}/${Folder_a_name}:${TAG} >> $(dirname $0)/../dist/images
```
Comment (orangedeng) [inline scripts/package]: this is needed

File: scripts/package [hunk 7]
```diff
@@ -15,12 +15,18 @@ cp ../bin/pipeline ./pipeline/
 
 TAG=${TAG:-${VERSION}${SUFFIX}}
 REPO=${REPO:-rancher}
+#Tag for dependent images
+PIPELINE_UI_TAG=v0.1.1
+JENKINS_BOOT_TAG=v0.1.1
```
Comment (biblesyme) [inline scripts/package]: https://github.com/biblesyme/jenkins-boot/releases
here is the release for jenkins-boot

File: scripts/package (hunk 5 above)
Comment (biblesyme) [reply scripts/package]: https://github.com/rancher/pipeline-ui/releases

Comment (gitlawr) [reply scripts/package]: I prefer using this bc we don't keep consistency with that before

File: scripts/package (hunk 6 above)
Comment (gitlawr) [reply scripts/package]: Updated

File: scripts/package (hunk 5 above)
Comment (biblesyme) [reply scripts/package]: We don't take it seriously before, I think. It may cause some confusion using different tag.

File: scripts/package [hunk 8]
```diff
@@ -15,12 +15,19 @@ cp ../bin/pipeline ./pipeline/
 
 TAG=${TAG:-${VERSION}${SUFFIX}}
 REPO=${REPO:-rancher}
+#Tag for dependent images
+PIPELINE_UI_TAG=v1.0.1
+JENKINS_BOOT_TAG=v1.0.0
+JENKINS_SLAVE_TAG=v0.1.0
```
Comment (deniseschannon) [inline scripts/package]: Are we hard-coding in the image tags for these images? 

```
rancher/pipeline-ui
rancher/pipeline-jenkins-slave
rancher/jenkins-boot
```

Is there a reason why it's called `jenkins-boot` and not `pipeline-jenkins-boot`?

So when we make a release, the only tag that will change will be for the `rancher/pipeline`?

Comment (gitlawr) [reply scripts/package]: Yes, they are hard-coded and need to be updated if corresponding dependent images are updated.
I will change to ‘pipeline-jenkins-boot’ to keep consistency.
When we make a release, we need to update the dependency tags if they are changed. And only ‘rancher/pipeline’ is affected by the git tag.

File: package/pipeline-jenkins-slave/Dockerfile [hunk 9]
```diff
@@ -9,7 +9,7 @@ RUN chmod +x /usr/local/bin/docker-wrapper && \
     chmod +x /usr/local/bin/wait-for-master
 
 # add tools for cicd
-ADD https://github.com/gitlawr/cihelper/releases/download/${PIPELINE_VERSION}/cihelper /usr/local/bin/cihelper
+ADD https://github.com/gitlawr/cihelper/releases/download/${CIHELPER_VERSION}/cihelper /usr/local/bin/cihelper
```
Comment (deniseschannon) [inline package/pipeline-jenkins-slave/Dockerfile]: Can we move this repo into rancher?

File: package/pipeline-jenkins-boot/Dockerfile [hunk 10]
```diff
@@ -1,5 +1,5 @@
 FROM busybox
-ENV BOOT_VERSION 1.0.0
+ARG BOOT_VERSION
 RUN wget -q -O /boot.tar.gz https://www.github.com/biblesyme/jenkins-boot/archive/v${BOOT_VERSION}.tar.gz
```
Comment (deniseschannon) [inline package/pipeline-jenkins-boot/Dockerfile]: Can we move this repo into rancher?

File: package/pipeline-jenkins-slave/Dockerfile [hunk 11]
```diff
@@ -1,6 +1,6 @@
 FROM rancher/jenkins-slave
 
-ENV PIPELINE_VERSION v1.0.0
+ENV CIHELPER_VERSION v1.0.0
```
Comment (deniseschannon) [inline package/pipeline-jenkins-slave/Dockerfile]: how come this isn't made an arg like in the other DockerFiles?

File: scripts/package [hunk 12]
```diff
@@ -15,12 +15,22 @@ cp ../bin/pipeline ./pipeline/
 
 TAG=${TAG:-${VERSION}${SUFFIX}}
 REPO=${REPO:-rancher}
+#Tag for dependent images
+PIPELINE_UI_VERSION=1.0.1
+PIPELINE_UI_TAG=v${PIPELINE_UI_VERSION}
```
Comment (deniseschannon) [inline scripts/package]: this seems redundant. can't we just use `v${PIPELINE_UI_VERSION}` for the tag section in the scripts below?

(same comment for the other `TAG` variables)

**Comment by gitlawr** [comment]
@deniseschannon  updated dependency to rancher

**Comment by deniseschannon** [comment]
LGTM after we transfer the jenkins-boot repo.

**Pull request merged**